cmd/argocd/commands/app.go

```diff
@@ -1849,6 +1851,9 @@ func NewApplicationSyncCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
   argocd app sync -l '!app.kubernetes.io/instance'
   argocd app sync -l 'app.kubernetes.io/instance notin (my-app,other-app)'
 
+  # Sync a multi-source application for specific revision of specific sources
+  argocd app manifests my-app --revisions 0.0.1 --source-positions 1 --revisions 0.0.2 --source-positions 2
```

Comment:
Hmm, I wonder if we should have named sources too. Maybe a followup PR, not worth stalling this one.

Reply:
There is an open issue opened for the same as a future enhancement. https://github.com/argoproj/argo-cd/issues/17731